src/components/editgrid/EditGrid.js

```diff
@@ -339,7 +339,23 @@ export default class EditGridComponent extends NestedComponent {
     });
     editRow.components = this.createRowComponents(editRow.data, rowIndex);
     this.checkRow(this.data, editRow);
-    this.redraw();
+    if (this.component.modal) {
+      this.addRowModal(rowIndex);
+    }
+    else {
+      this.redraw();
+    }
+  }
+
+  addRowModal(rowIndex) {
+    const formComponents =  this.ce('div');
+    formComponents.innerHTML = this.renderComponents(this.editRows[rowIndex].components);
+    const dialog = this.component.modal ? this.createModal(formComponents) : undefined;
+    dialog.refs.dialogContents.appendChild( this.ce('button', {
```

Comment:
This would also be ```undefined``` if this.component.modal is false. Should have an if statement around this.